lib/editor/components/pattern/PatternStopCard.js

```diff
@@ -254,42 +254,29 @@ class PatternStopContents extends Component<Props, State> {
     updatePatternStops(activePattern, patternStops)
   }
 
-  _onPickupChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
-    const newPickup: number = parseInt(evt.target.value)
-    const {activePattern, index, updatePatternStops} = this.props
-    const patternStops = [...activePattern.patternStops]
-
-    patternStops[index].pickupType = newPickup
-    this.setState({update: true})
-    updatePatternStops(activePattern, patternStops)
-  }
+  _onPickupOrDropOffChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    const selectOptionType: string = evt.target.id
+    const selectedOptionValue: number = parseInt(evt.target.value)
 
-  _onDropOffChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
-    const newDropOff: number = parseInt(evt.target.value)
     const {activePattern, index, updatePatternStops} = this.props
     const patternStops = [...activePattern.patternStops]
 
-    patternStops[index].dropOffType = newDropOff
-    this.setState({update: true})
-    updatePatternStops(activePattern, patternStops)
-  }
-
-  _onContinuousPickupChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
-    const newPickup: number = parseInt(evt.target.value)
-    const {activePattern, index, updatePatternStops} = this.props
-    const patternStops = [...activePattern.patternStops]
-
-    patternStops[index].continuousPickup = newPickup
-    this.setState({update: true})
-    updatePatternStops(activePattern, patternStops)
-  }
-
-  _onContinuousDropOffChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
-    const newDropOff: number = parseInt(evt.target.value)
-    const {activePattern, index, updatePatternStops} = this.props
-    const patternStops = [...activePattern.patternStops]
-
-    patternStops[index].continuousDropOff = newDropOff
+    switch (selectOptionType) {
+      case 'continuous_pick-up':
```

Comment:
If `selectOptionType` were the same as the field in `patternStops[index]` we wanted to modify, you wouldn't need this `switch` block, or does flow require it?